hawkbit-core/src/main/java/org/eclipse/hawkbit/repository/RolloutFields.java

```diff
@@ -25,12 +29,35 @@ public enum RolloutFields implements FieldNameProvider {
     /**
      * The id field.
      */
-    ID("id");
+    ID("id"),
+    /**
+     * The status field.
+     */
+    STATUS("status"),
+    /**
+     * The Distribution set field.
+     */
+    DISTRIBUTIONSET("distributionSet", DistributionSetFields.ID.getFieldName(),
+            DistributionSetFields.NAME.getFieldName(), DistributionSetFields.VERSION.getFieldName(),
+            DistributionSetFields.TYPE.getFieldName());
 
     private final String fieldName;
 
+    private List<String> subEntityAttributes;
+
     private RolloutFields(final String fieldName) {
         this.fieldName = fieldName;
+        this.subEntityAttributes = Collections.emptyList();
+    }
+
+    private RolloutFields(final String fieldName, final String... subEntityAttributes) {
+        this.fieldName = fieldName;
+        this.subEntityAttributes = Arrays.asList(subEntityAttributes);
+    }
+
+    @Override
+    public List<String> getSubEntityAttributes() {
+        return subEntityAttributes;
```

Comment:
This way the returned list elements could be modified which modifies the enum which shall be immutable.
Better return Collections.unmodifiableList(subEntityAttributes)